main.go

```diff
@@ -105,22 +135,75 @@ func main() {
 		glog.Fatalf("Failed to create a kubernetes client: %v", err)
 	}
 
+	if !opts.noHealth {
+		go func() {
+			// create http server to expose health status
+			r := mux.NewRouter()
+			r.HandleFunc("/health", health.StatusHandler)
+			srv := &http.Server{
+				Handler:     r,
+				Addr:        "0.0.0.0:8080",
+				ReadTimeout: 15 * time.Second,
+			}
+			glog.Fatal(srv.ListenAndServe())
+		}()
+	}
+
+	if opts.lock {
+		health.RegisterPeriodicThresholdFunc("consul", time.Second*5, 3, consulCheck)
+		health.RegisterPeriodicThresholdFunc("kubernetes", time.Second*5, 3, kubernetesCheck)
```

Comment:
I think we want to check k8s and consul connection health whether or not we want to acquire a consul lock, don't we ?